zerver/webhooks/jira/view.py

```diff
@@ -271,7 +271,7 @@ def api_jira_webhook(request: HttpRequest, user_profile: UserProfile,
                      payload: Dict[str, Any]=REQ(argument_type='body')) -> HttpResponse:
 
     event = get_event_type(payload)
-    subject = get_issue_subject(payload)
+    subject = get_issue_subject(payload)  # type: Optional[str]
```

Comment:
This I think goes a little deeper; to `get_in`, which is defined as:
```
def get_in(payload: Dict[str, Any], keys: List[str], default: str='') -> Any:
    try:
        for key in keys:
            payload = payload[key]
    except (AttributeError, KeyError, TypeError):
        return default
    return payload
```

The following callers state the return type as `str`, but then elsewhere variables assigned to `get_in` are used as a `Dict`.  This seems to happen interchangeably, not sure what's going on with that.

```
def get_issue_id(payload: Dict[str, Any]) -> str:
    return get_in(payload, ['issue', 'key'])

def get_issue_title(payload: Dict[str, Any]) -> str:
    return get_in(payload, ['issue', 'fields', 'summary'])

def get_issue_subject(payload: Dict[str, Any]) -> str:
    return u"{}: {}".format(get_issue_id(payload), get_issue_title(payload))
```
A few points:
* `u`, I believe, is now redundant.
* `default` in `get_in` is never overridden.
* It looks like `subject` will never return `None`, as `get_issue_id` and `get_issue_title` either return the `payload` or `''`, so subject will wither be `"{}: {}"` or `"{<id>: <title>}"`

I think maybe just completely remove the `if subject is None` check for now?  But it does seem like something more is going on here.

Reply:
`u` is indeed unnecssary.  And yes, removing that check is the right fix; it's leftover from a refactor.